File: app/src/main/java/de/sneakpeek/data/MovieInfoCredits.java
package de.sneakpeek.data;


import android.os.Parcel;
import android.os.Parcelable;

import com.squareup.moshi.Json;

import java.util.List;

public class MovieInfoCredits implements Parcelable {

    public final static Parcelable.Creator<MovieInfoCredits> CREATOR = new Creator<MovieInfoCredits>() {

        public MovieInfoCredits createFromParcel(Parcel in) {
            MovieInfoCredits instance = new MovieInfoCredits();
            in.readList(instance.cast, (MovieInfoCast.class.getClassLoader()));
            in.readList(instance.crew, (MovieInfoCrew.class.getClassLoader()));
            return instance;
        }

        public MovieInfoCredits[] newArray(int size) {
            return (new MovieInfoCredits[size]);
        }

    };

    @Json(name = "cast") public List<MovieInfoCast> cast;
    @Json(name = "crew") public List<MovieInfoCrew> crew;

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(cast);
        dest.writeValue(crew);
    }

    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/de/sneakpeek/data/MovieInfoCredits.kt
package de.sneakpeek.data


import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class MovieInfoCredits(
        @Expose @SerializedName("cast") var cast: List<MovieInfoCast>?,
        @Expose @SerializedName("crew") var crew: List<MovieInfoCrew>?) : Parcelable {

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<MovieInfoCredits> = object : Parcelable.Creator<MovieInfoCredits> {
            override fun createFromParcel(source: Parcel): MovieInfoCredits = MovieInfoCredits(source)
            override fun newArray(size: Int): Array<MovieInfoCredits?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.createTypedArrayList(MovieInfoCast.CREATOR),
            source.createTypedArrayList(MovieInfoCrew.CREATOR)
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeTypedList(cast)
        dest.writeTypedList(crew)
    }
}


